lib/membrane_rtc_engine/endpoints/webrtc/connection_prober.ex

```diff
@@ -0,0 +1,113 @@
+
+  @spec start_link() :: GenServer.on_start()
+  def start_link(), do: GenServer.start_link(__MODULE__, [], [])
+
+  @impl GenServer
+  def init(_opts) do
+    state = %{
+      bandwidth_estimation: nil,
+      bitrate_timer: nil,
+      estimation_timestamp: 0,
+      bytes_sent: 0,
+      track_receivers: Qex.new()
+    }
+
+    {:ok, state}
+  end
+
+  @impl GenServer
+  def handle_cast({:bandwidth_estimation, estimation}, state) do
+    if state.bitrate_timer, do: :timer.cancel(state.bitrate_timer)
+
+    {:ok, timer} = :timer.send_interval(10, :check_bytes_sent)
+
+    state = %{
+      state
+      | # TODO: rethink how target bandwidth should be calculated - + 200kbps seems potentially problematic
+        bandwidth_estimation: (estimation + 200_000) / 8,
+        estimation_timestamp: get_timestamp(),
+        bitrate_timer: timer,
+        bytes_sent: 0
+    }
+
+    {:noreply, state}
+  end
+
+  @impl GenServer
+  def handle_cast({:buffer_sent, size}, state) do
+    state = Map.update!(state, :bytes_sent, &(&1 + size))
+    {:noreply, state}
+  end
+
+  @impl GenServer
+  def handle_cast({:register_track_receiver, tr}, state) do
+    state = Map.update!(state, :track_receivers, &Qex.push_front(&1, tr))
+    {:noreply, state}
+  end
+
+  @impl GenServer
+  def handle_info(:check_bytes_sent, state) do
+    use Numbers, overload_operators: true
+
+    now = get_timestamp()
+    elapsed_time_in_s = Time.as_seconds(now - state.estimation_timestamp)
+    expected_bytes = elapsed_time_in_s * state.bandwidth_estimation
+    missing = expected_bytes - state.bytes_sent
+
+    state =
+      if missing > 0 do
+        # Send paddings
+
+        no_padding_packets =
+          missing
+          |> Ratio.new(@padding_packet_size)
+          |> Ratio.ceil()
+
+        state
+        |> send_padding_packets(no_padding_packets)
+        |> Map.update!(:bytes_sent, &(&1 + no_padding_packets * @padding_packet_size))
+      else
+        state
+      end
+
+    {:noreply, state}
+  end
+
+  ## Public API
+
+  @spec update_bandwidth_estimation(pid(), number()) :: :ok
+  def update_bandwidth_estimation(prober, estimation),
+    do: GenServer.cast(prober, {:bandwidth_estimation, estimation})
+
+  @spec buffer_sent(pid(), Buffer.t()) :: :ok
+  def buffer_sent(prober, %Buffer{payload: payload}),
+    do: GenServer.cast(prober, {:buffer_sent, byte_size(payload)})
+
+  @spec register_track_receiver(pid(), pid()) :: :ok
+  def register_track_receiver(prober, tr \\ self()),
+    do: GenServer.cast(prober, {:register_track_receiver, tr})
+
+  ## Helper functions
+  defp get_timestamp(), do: System.monotonic_time(:nanosecond)
+
+  defp send_padding_packets(state, 0), do: state
+
+  defp send_padding_packets(state, no_packets) do
+    Enum.reduce(1..no_packets, state, fn _i, state ->
+      # We need to select a track receiver in such a way that each one sends an equal amount of packets to create
+      # => Round Robin
```

Comment:
Do we? We can't we send everything to one track receiver?

```suggestion
      # We need to select a track receiver in such a way that each one sends an equal amount of packets
      # => Round Robin
```

Reply:
I don't think there is any reason to overload a single track receiver while other ones are chilling. Especially in the current configuration, balancing even the sequence number space usage across all track receivers seems like a good idea.

Reply:
Yeap but there is a difference between `we need (because e.g. borwser requires this)` and `it's a good idea`. Let's adjust comment